src/engine/sparqlExpressions/NaryExpression.cpp

```diff
@@ -30,6 +32,30 @@ SparqlExpression::Ptr makeDistExpression(SparqlExpression::Ptr child1,
   return std::make_unique<DistExpression>(std::move(child1), std::move(child2));
 }
 
+std::optional<GeoFunctionCall> getGeoFunctionExpressionParameters(
+    const SparqlExpression& expr) {
+  // TODO<ullingerc> After merge of #1938 handle distance unit and
+  // metricDistance function
+
+  // Currently only DistExpression
+  auto distExpr = dynamic_cast<const DistExpression*>(&expr);
+  if (distExpr == nullptr) {
+    return std::nullopt;
+  }
+  auto p1 =
+      dynamic_cast<LiteralExpression<Variable>*>(distExpr->children()[0].get());
+  if (p1 == nullptr) {
+    return std::nullopt;
+  }
```

Comment:
As we need this pattern often, SparqlExpression has a `getVariableOrNullopt` function.